qbraid/runtime/origin/provider.py

```
@@ -0,0 +1,153 @@
+    try:
+        chip_info = backend.chip_info()
+        return chip_info.get_basic_gates()
+    except Exception:
+        logger.debug("Unable to determine basis gates from chip info", exc_info=True)
+        return None
+
+
+class OriginProvider(QuantumProvider):
+    """OriginQ QCloud provider class."""
+
+    def __init__(self, api_key: Optional[str] = None):
```

Ruff (0.15.9)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)